### PR TITLE
CASSANDRA-17753: Include GitSHA in nodetool version output

### PR DESCRIPTION
**Pull request by aratno**
Jira: https://issues.apache.org/jira/browse/CASSANDRA-17753
CircleCI: https://app.circleci.com/pipelines/github/aratno/cassandra?branch=CASSANDRA-17753-nodetool-version-gitsha

```bash
# When Cassandra is built with a clean Git index
$ ./bin/nodetool version
ReleaseVersion: 4.1-alpha2-SNAPSHOT
GitSHA: 5fce07e2f1e2e0b9cf7d82adfd8c0b21993f8672

# When Cassandra is built with a dirty Git index
$ ./bin/nodetool version
ReleaseVersion: 4.1-alpha2-SNAPSHOT
GitSHA: 5fce07e2f1e2e0b9cf7d82adfd8c0b21993f8672-dirty
```

### CODE REVIEW

File: conf/cassandra.yaml [hunk 1]
```diff
@@ -1791,3 +1791,4 @@ drop_compact_storage_enabled: false
 #    heartbeat_file: /var/lib/cassandra/data/cassandra-heartbeat
 #    excluded_keyspaces: # comma separated list of keyspaces to exclude from the check
 #    excluded_tables: # comma separated list of keyspace.table pairs to exclude from the check
+
```
Comment (maedhroz) [inline conf/cassandra.yaml:1879]: What's up w/ the newline here?

**Review by maedhroz — APPROVED** [review]

File: conf/cassandra.yaml (hunk 1 above)
Comment (aratno) [reply conf/cassandra.yaml:1879]: Without it, ant adds it, and dirties the build every time.

**Review by frankgh — APPROVED** [review]
+1

File: CHANGES.txt [hunk 2]
```diff
@@ -15,6 +15,7 @@
  * Add new CQL function maxWritetime (CASSANDRA-17425)
  * Add guardrail for ALTER TABLE ADD / DROP / REMOVE column operations (CASSANDRA-17495)
  * Rename DisableFlag class to EnableFlag on guardrails (CASSANDRA-17544)
+ * `nodetool version` now includes GitSHA
```
Comment (michaelsembwever) [inline CHANGES.txt]: CHANGES list is reverse chronological (you put new entries in at the top, not the bottom)

**Comment by michaelsembwever** [comment]
suggestions/requests raised in the ticket

File: NEWS.txt [hunk 3]
```diff
@@ -70,6 +70,8 @@ New features
       - Whether ALTER TABLE commands are allowed to mutate columns
       - Whether SimpleStrategy is allowed on keyspace creation or alteration
       - Maximum replication factor
+    - `nodetool version` now includes the Git SHA the project was built with, and whether it was built with a dirty Git
+      index; build now depends on a local Git installation
```
Comment (michaelsembwever) [inline NEWS.txt]: i think we can remove this (it's not NEWS worthy enough)

File: src/java/org/apache/cassandra/tools/nodetool/Version.java [hunk 4]
```diff
@@ -19,15 +19,23 @@
 
 import io.airlift.airline.Command;
 
+import io.airlift.airline.Option;
```
Comment (michaelsembwever) [inline src/java/org/apache/cassandra/tools/nodetool/Version.java]:
```suggestion
import io.airlift.airline.Option;

```

**Review by michaelsembwever — APPROVED** [review]
two small nit comments, otherwise looking great and +1. thanks @aratno

**Comment by aratno** [comment]
Thanks @michaelsembwever  - all changes requested are now addressed. Do you want me to merge in trunk and resolve conflicts, squash, or anything else to make this easier to commit?

**Comment by michaelsembwever** [comment]
Yes, please squash to one commit. Are you familiar with our commit message format? (if not, the committer will adjust it accordingly)

There's also one minor edit to make, the CHANGES.txt line need to finish with ` (CASSANDRA-17753)`

**Comment by aratno** [comment]
Should be all good now - did I do it right @michaelsembwever?

**Review by michaelsembwever — APPROVED** [review]

**Comment by michaelsembwever** [comment]
ci-cassandra: https://ci-cassandra.apache.org/job/Cassandra-devbranch-artifacts/1749/ 

LGTM 👍

**Comment by michaelsembwever** [comment]
latest changed LGTM! thanks!

https://ci-cassandra.apache.org/job/Cassandra-devbranch/1886/

**Comment by michaelsembwever** [comment]
>  https://ci-cassandra.apache.org/job/Cassandra-devbranch/1886/ 

something is up with the dtest-upgrade… (took a re-run of it here too: https://ci-cassandra.apache.org/job/Cassandra-devbranch-dtest-upgrade/1177/ )
haven't had time to investigate it… but wondering if this patch needs to be rebased off latest trunk…

**Comment by aratno** [comment]
Just rebased, looks like the Paxos state issue impacting test clean-up across the board.

> Unexpected error found in node logs (see stdout for full details). Errors: [[node1] 'ERROR [main] 2022-08-20 10:40:54,861 CassandraDaemon.java:896 - Exception encountered during startup\njava.lang.ExceptionInInitializerError: null\n\tat org.apache.cassandra.service.paxos.PaxosState.initializeTrackers(PaxosState.java:116)

**Comment by michaelsembwever** [comment]
new round: https://ci-cassandra.apache.org/blue/organizations/jenkins/Cassandra-devbranch/detail/Cassandra-devbranch/1900/pipeline

**Comment by aratno** [comment]
Another CircleCI run: https://app.circleci.com/pipelines/github/aratno/cassandra?status=none&status=running&branch=CASSANDRA-17753-nodetool-version-gitsha

**Pull request closed without merging**